### PR TITLE
Process NOT NULL in gatherConditions

### PR DESCRIPTION
**Pull request by githabas**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no

#### Summary
Adds possibility to specify WHERE IS NOT NULL in conditions.

### CODE REVIEW

**Review by SenseException — CHANGES_REQUESTED** [review]
Thank your for your contribution. These changes are breaking the tests with the error message 
`strtoupper() expects parameter 1 to be string, object given`. There are also a few coding style fixes needed.

If you have questions about your PR or the build feel free to ask. Can you please describe your use case for your changes?

File: lib/Doctrine/DBAL/Connection.php [hunk 1]
```diff
@@ -615,9 +615,14 @@ private function addIdentifierCondition(
...
 
         foreach ($identifier as $columnName => $value) {
-            if ($value === null) {
-                $conditions[] = $platform->getIsNullExpression($columnName);
-                continue;
+            if (!is_array($value)) {
+                if ($value === null) {
+                    $conditions[] = $platform->getIsNullExpression($columnName);
+                    continue;
+                } elseif (strtoupper($value) === 'NOT NULL') {
```
Comment (morozov) [inline lib/Doctrine/DBAL/Connection.php]: How is it supposed to work with a string value "NOT NULL"?

Comment (githabas) [reply lib/Doctrine/DBAL/Connection.php]: $this->connection->update('connections', ['timezone' => 'my zone'], ['timezone' => 'NOT NULL']);

Comment (morozov) [reply lib/Doctrine/DBAL/Connection.php]: What if I want to check that the value is not equal to the string "NOT NULL"?

Comment (githabas) [reply lib/Doctrine/DBAL/Connection.php]: Ok it's bad idea. I think we should rather look at columnName

**Pull request closed without merging**